From a71a688df381dd8ccc1a1aa476ff6f00e7a2e229 Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Thu, 3 Dec 2015 17:07:42 +0800
Subject: [PATCH 06/13] atomisp2: WORKAROUND assign default OB2 param for BTNS
 project

On BTNS, with imx227 or OV8858, without the default OB2 setting
{1024,1024,1024,1024}, the captured image looks pink tint.
So added this workaround to make image looks better.

In the future when AIQ and ISP2.7 tunning is available, remove it
and AIQ will take care of OB2 setting.

Change-Id: I3094ea4c46d4a898427988afbbc93afa7825aa70
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-13009
Original-Tracked-On: #H1604058414
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/87948
Reviewed-by: HaiqiangX Yao <haiqiangx.yao@intel.com>
Build: Dessert M BXT RVP (SC)
Build: Dessert M COHO (SC)
Approver: Tian Shu Qiu <tian.shu.qiu@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/450387
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c   | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 17520af..310ed68 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -3214,6 +3214,17 @@ void atomisp_apply_css_parameters(
 	if (css_param->update_flag.dvs_6axis_config)
 		atomisp_css_set_dvs_6axis(asd, css_param->dvs_6axis);
 
+	/* Workaround for BTNS, apply default OB2 setting for IMX227.
+	   before AIQ for ISP2.7 is available,	we have to keep such
+	   workaround for a reasonable image color
+	 */
+#ifdef CONFIG_PLATFORM_BTNS
+	{
+		static struct atomisp_css_ob2_config imx227_ob2_config =
+			{1024, 1024, 1024, 1024};
+		atomisp_css_set_ob2_config(asd, &imx227_ob2_config);
+	}
+#endif
 	atomisp_css_set_isp_config_id(asd, css_param->isp_config_id);
 	/*
 	 * These configurations are on used by ISP1.x, not for ISP2.x,
-- 
1.9.1

